dali/operators/generic/transpose/transpose_test.cc

```diff
@@ -182,26 +238,178 @@ void TransposeVerify(const testing::TensorListWrapper& input,
 TEST_P(TransposeTestRank3, TransposeRank3) {
   auto args = GetParam();
   testing::TensorListWrapper tlout;
-  this->RunTest(GetInput(3).get(), tlout, args, TransposeVerify);
+  auto mask = args.at(testing::ArgumentKey(kOnesMask)).GetValue<unsigned int>();
+  auto uniform = args.at(testing::ArgumentKey(kUniform)).GetValue<bool>();
+  this->RunTest(GetInput(3, mask, uniform).get(), tlout, args, TransposeVerify);
 }
 
 TEST_P(TransposeTestRank4, TransposeRank4) {
   auto args = GetParam();
   testing::TensorListWrapper tlout;
-  this->RunTest(GetInput(4).get(), tlout, args, TransposeVerify);
+  auto mask = args.at(testing::ArgumentKey(kOnesMask)).GetValue<unsigned int>();
+  auto uniform = args.at(testing::ArgumentKey(kUniform)).GetValue<bool>();
+  this->RunTest(GetInput(4, mask, uniform).get(), tlout, args, TransposeVerify);
 }
 
 TEST_P(TransposeTestRank5, TransposeRank5) {
   auto args = GetParam();
   testing::TensorListWrapper tlout;
-  this->RunTest(GetInput(5).get(), tlout, args, TransposeVerify);
+  auto mask = args.at(testing::ArgumentKey(kOnesMask)).GetValue<unsigned int>();
+  auto uniform = args.at(testing::ArgumentKey(kUniform)).GetValue<bool>();
+  this->RunTest(GetInput(5, mask, uniform).get(), tlout, args, TransposeVerify);
 }
 
 INSTANTIATE_TEST_SUITE_P(TransposeRank3Suite, TransposeTestRank3,
-                        ::testing::ValuesIn(cartesian(devices, GetPermutations(3))));
+                         ::testing::ValuesIn(cartesian(devices, GetPermutations(3), GetOneMasks(3),
+                                                       uniform)));
 INSTANTIATE_TEST_SUITE_P(TransposeRank4Suite, TransposeTestRank4,
-                        ::testing::ValuesIn(cartesian(devices, GetPermutations(4))));
+                         ::testing::ValuesIn(cartesian(devices, GetPermutations(4), GetOneMasks(4),
+                                                       uniform)));
 INSTANTIATE_TEST_SUITE_P(TransposeRank5Suite, TransposeTestRank5,
-                        ::testing::ValuesIn(cartesian(devices, GetPermutations(5))));
+                         ::testing::ValuesIn(cartesian(devices, GetPermutations(5),
+                                                       one_masks_reduced, uniform)));
+
+TEST(TransposeTest, PrepareArgumentsNoOnes) {
+  using dtype = SmallVector<int, 6>;
```

Comment:
something more descriptive?
```suggestion
  using array = SmallVector<int, 6>;
```
```suggestion
  using intvec = SmallVector<int, 6>;
```

Reply:
Done